### PR TITLE
statscache: support multi-level evict

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/45367

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by winoros — APPROVED** [review]

File: statistics/handle/cache/internal/lfu/lfu_cache.go [hunk 1]
```diff
@@ -105,11 +108,30 @@ func (s *LFU) Values() []*statistics.Table {
...
+	for _, column := range table.Columns {
+		DropEvicted(column)
+	}
+	for _, indix := range table.Indices {
+		DropEvicted(indix)
+	}
+	after := table.MemoryUsage().TotalTrackingMemUsage()
+	// why add before again? because the cost will be subtracted in onExit.
+	// in fact, it is  -(before - after) + after = after + after - before
+	s.cost.Add(2*after - before)
```
Comment (qw4990) [inline statistics/handle/cache/internal/lfu/lfu_cache.go:134]: Any test on this?

**Review by qw4990 — APPROVED** [review]

File: statistics/handle/cache/internal/lfu/lfu_cache.go (hunk 1 above)
Comment (hawkingrei) [reply statistics/handle/cache/internal/lfu/lfu_cache.go:134]: It exists in the existing test.

**Pull request merged**